### PR TITLE
refac(watt): refactor watt-field to use signals + more

### PR DESCRIPTION
**Pull request by ManBearTM**
Closes https://github.com/Energinet-DataHub/greenforce-frontend/issues/3178

I also had to make the component subscribe to statusChanges, since otherwise the `isRequired` state would not be properly updated when dynamically changing form control validators (yes that is a use case now 😄). Also meant that I could change the component to use OnPush for change detection.

### CODE REVIEW

File: libs/watt/src/lib/components/field/watt-field.component.ts [hunk 1]
```diff
@@ -14,88 +14,96 @@
...
+  wrapper = viewChild.required<ElementRef>('wrapper');
 
-  @HostBinding('class.watt-field--invalid')
-  get _hasError() {
-    return (
-      (this.control?.status === 'INVALID' && !!this.control?.touched) ||
-      (this._formGroupDirective?.submitted && this.control?.status === 'INVALID')
+  constructor() {
+    const status$ = toObservable(this.control).pipe(
+      filter((control) => control !== null),
```
Comment (mimse) [inline libs/watt/src/lib/components/field/watt-field.component.ts:91]: Question: Should there be a unsubscribe somewhere? Or a takeUntilDestoryed?

**Review by mimse — APPROVED** [review]

**Review by mimse — COMMENTED** [review]
Praise: Good job

File: libs/watt/src/lib/components/field/watt-field.component.ts (hunk 1 above)
Comment (ManBearTM) [reply libs/watt/src/lib/components/field/watt-field.component.ts:91]: No need, it completes itself:
https://github.com/angular/angular/blob/main/packages/core/rxjs-interop/src/to_observable.ts#L65

Comment (mimse) [reply libs/watt/src/lib/components/field/watt-field.component.ts:91]: Perfect!

**Pull request merged**